docs/concepts/bitswap.md

```diff
@@ -11,4 +10,54 @@ related:
 
 # Bitswap
 
-<ContentStatus />
+Bitswap is a core module of IPFS for exchanging blocks of data. It directs the requesting and sending of blocks to and from other peers in the network. Bitswap is a _message-based protocol_ where all messages contain [wantlists](#wantlist) or blocks. Bitswap has a [Go implementation](https://github.com/ipfs/go-bitswap) and a [JavaScript implementation](https://github.com/ipfs/js-ipfs-bitswap).
+
+Bitswap has two main jobs:
+- to acquire blocks requested by the client from the network.
+- to send blocks in its possession to other peers who want them.
+
+### Wantlist
+
+A wantlist is a data structure that an IPFS node sends to the network to tell peers which blocks it wants. The wantlist contains [CIDs](https://docs-beta.ipfs.io/concepts/content-addressing/) (content-addressed identifiers that refer to particular blocks desired) along with priority and cancellation information.
+
+### Message
+
+A single Bitswap message may contain any of the following content:
+
+-  **The sender’s wantlist.** This wantlist may either be the sender’s complete wantlist or just the changes to the sender’s wantlist that the receiver needs to know.
+-  **Data blocks.** These are blocks that have been requested (i.e., blocks in the sender’s wantlist, as far as the receiver is aware at the time of message receipt).
+
+### Session
+
+IPFS content blocks are often connected to each other through a [Merkle DAG](https://docs-beta.ipfs.io/concepts/merkle-dag/). If a sender's block requests are related, a mechanism called a _Bitswap session_ can optimize block requests sent to other peers. This mechanism can increase transfer speed and reduce the number of duplicate blocks on the network. Active peers are favored, as they are tracked relative to them having the requested blocks and how quickly they respond.
+
+### Provider/Providing
+
+The provider system is responsible for announcing and reannouncing to the IPFS network that a node has content. If a node cannot find content with connected peers, it will query a content routing system (usually a form of [DHT](https://docs-beta.ipfs.io/concepts/dht/)) to locate a peer with the content. Bitswap routes these requests through the `ProviderQueryManager` system. As a node receives blocks, it announces them on the provided content routing system.
+
+### Deeper Examination
+
+A diagram showing the primary tasks of Bitswap and underlying components:
+
+![Bitswap task and component diagram](https://github.com/ipfs/go-bitswap/blob/master/docs/go-bitswap.png "Go-Bitswap")
```

Comment:
I am not sure if this is up to date or will stay up to date in the near future (cc. @dirkmc)

Reply:
Also, should note that this is specific to the go-bitswap implementation.

Reply:
This is going to change with the next release of go-bitswap, and is not so relevant for other implementations of Bitswap. I'm not sure how useful it is for the audience that this document is aimed at to know about the architectural details of how Bitswap works. Probably the broader, conceptual picture outlined in the previous section is sufficient.